### PR TITLE
Add indexes for status and request

### PR DESCRIPTION
**Pull request by tyll**

### CODE REVIEW

**Comment by ryanlerch** [comment]
jenkies test

**Comment by ryanlerch** [comment]
@centos-ci test this please

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+Add indexes for status and request
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: pydocstyle wants a period on the end of this sentence.

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
+# revision identifiers, used by Alembic.
+revision = '2616c86d8ac6'
+down_revision = '95ce24bed77a'
+
+
+def upgrade():
+    ### commands auto generated by Alembic - please adjust! ###
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: pydocstyle wants a docblock here. How about this:

```
"""
Add indices to the request and status columns on the update table, and add a compound uniqueness constraint on the Package's name and type.
"""
```

Similarly, pydocstyle wants a docblock on the ```downgrade()``` function below and we can probably just reverse the above sentence to say its removing those indices/constraints.

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    op.create_index(op.f('ix_updates_status'), 'updates', ['status'], unique=False)
+    ### end Alembic commands ###
+
+
+def downgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.drop_index(op.f('ix_updates_status'), table_name='updates')
+    op.drop_index(op.f('ix_updates_request'), table_name='updates')
+    op.drop_constraint('packages_name_and_type_key', 'packages', type_='unique')
+    ### end Alembic commands ###
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: I suggest removing the autogenerated Alembic comments from this file.

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+Add indexes for status and request
+
+Revision ID: 2616c86d8ac6
+Revises: 95ce24bed77a
+Create Date: 2017-10-26 20:45:23.152139
+"""
+from alembic import op
+import sqlalchemy as sa
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: This import isn't used so we should drop it.

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: Let's remove one of these three blank lines so there are two, for PEP-8.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
This looks good! We just have a few style checkers that are upset (you can see these buried in the Jenkins output), but I've commented on all the things the checkers were upset about below. The migration looks good and seems to work.

File: bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+# revision identifiers, used by Alembic.
+revision = '2616c86d8ac6'
+down_revision = '95ce24bed77a'
+
+
+def upgrade():
+    ### commands auto generated by Alembic - please adjust! ###
+    op.create_unique_constraint('packages_name_and_type_key', 'packages', ['name', 'type'])
```
Comment (bowlofeggs) [inline bodhi/server/migrations/versions/2616c86d8ac6_add_indexes_for_status_and_request.py]: I added this in https://github.com/fedora-infra/bodhi/pull/2039, so we don't need it here or in ```downgrade()``` below.

**Comment by bowlofeggs** [comment]
Hello @tyll, do you still want to work on this?

**Comment by tyll** [comment]
On Wed, Jan 10, 2018 at 04:59:45PM +0000, Randy Barlow wrote:
> Hello @tyll, do you still want to work on this?

Thank you for your patience and the feedback. I updated the commit.

**Comment by bowlofeggs** [comment]
jenkies test

**Comment by bowlofeggs** [comment]
Hi @tyll!

I am going to force push your branch to fix up a couple things to get this ready to merge. Thanks for the patch!

**Comment by bowlofeggs** [comment]
jenkies test

**Review by bowlofeggs — APPROVED** [review]
Thanks!

**Comment by bowlofeggs** [comment]
The patch for this feature has been included in the [```3.3.0b0``` build](https://bodhi.stg.fedoraproject.org/docs/user/release_notes.html), which is currently deployed to [staging](https://bodhi.stg.fedoraproject.org).

**Pull request merged**